gene/delete.go

```diff
@@ -339,3 +206,255 @@ func deleteStable(seq *[]dna.Base, delStart int, delEnd int) error {
 	copy((*seq)[delStart:cap(*seq)], (*seq)[delEnd:cap(*seq)]) // shift bases in slice to remove deleted bases
 	return nil
 }
+
+// deleteUpdateGenome updates the genomic sequence upon deletion
+func deleteUpdateGenome(g *Gene, genomeIndexStartPos int, genomeIndexEndPos int) {
+	deletionLen := genomeIndexEndPos - genomeIndexStartPos
+	copy(g.genomeSeq[genomeIndexStartPos:], g.genomeSeq[genomeIndexEndPos:])
+	g.genomeSeq = g.genomeSeq[:len(g.genomeSeq)-deletionLen]
+
+	// Update startPos if necessary
+	if genomeIndexStartPos == 0 {
+		g.startPos += genomeIndexEndPos
+	}
+}
+
+// deleteUpdateCds updates the coding sequence upon deletion
+func deleteUpdateCds(g *Gene, genomeIndexStartPos int, genomeIndexEndPos int) (deletedCodingBases int, err error) {
+	var cdsIndexesToDelete []int
+	var codingDelStart, codingDelEnd int = -1, -1
+	deletionLen := genomeIndexEndPos - genomeIndexStartPos
+	for i := 0; i < len(g.cdsStarts); i++ {
+
+		switch {
+		// if cds is before deletion
+		case genomeIndexStartPos > g.cdsEnds[i]:
+			codingDelStart = int(g.featureArray[g.cdsEnds[i]]) + 1
+
+		// If cds exon is fully deleted
+		case genomeIndexStartPos <= g.cdsStarts[i] && genomeIndexEndPos > g.cdsEnds[i]:
+			// mark cds for deletion
+			cdsIndexesToDelete = append(cdsIndexesToDelete, i)
```

Comment:
This is the only place I see cdsIndexesToDelete updated, but it seems like we would also need to remove some indices if we have a deletion that starts in one cds and then ends in another.